### PR TITLE
Update RawScrollbar to support the track

### LINKED ISSUE #96936: Clean up Scrollbar API
Over a year ago, we completely re-vamped scrollbars while avoiding making any breaking changes.
Once they reached stable, the API was further iterated on as users started adopting the new scrollbar widgets.
There have been a lot of changes made, and I think we should now tidy up the API - especially with the ability to write dart fixes, it will be much easier to clean everything up nicely.

I thought I could do this in one go (https://github.com/flutter/flutter/pull/96935), but the change grew to be pretty large, so I am going to split it up and track it here.

Affected classes:

- RawScrollbar
- Scrollbar (_MaterialScrollbar)
- CupertinoScrollbar
- ScrollbarThemeData

- [x] Add thumbVisibility across all
  - https://github.com/flutter/flutter/pull/96957 
- [x] Deprecate isAlwaysShown across all
  - https://github.com/flutter/flutter/pull/96957
- [ ] Add trackVisibility, trackColor, trackBorderColor to RawScrollbar
  - https://github.com/flutter/flutter/pull/97335 
- [x] Deprecate showTrackOnHover in material library (widget & theme)
  - https://github.com/flutter/flutter/pull/97173 
- [x] Deprecate hoverThickness in material library (widget & theme)
  - https://github.com/flutter/flutter/pull/97173

### PR DESCRIPTION
**Pull request by Piinks**
Last of several scrollbar clean up PRs to prepare for desktop scrollbars.
Original PR was #96935, but it was becoming too large
Fixes https://github.com/flutter/flutter/issues/96936

This PR adds track features as part of the RawScrollbar base class. This will make it easier for subclasses to have a track. It also ensures that if a track is visible, the scrollbar thumb is visible too.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/scrollbar.dart [hunk 1]
```diff
@@ -350,7 +350,7 @@ class _MaterialScrollbarState extends RawScrollbarState<_MaterialScrollbar> {
     final Color onSurface = _colorScheme.onSurface;
     final Brightness brightness = _colorScheme.brightness;
     return MaterialStateProperty.resolveWith((Set<MaterialState> states) {
-      if (_trackVisibility.resolve(states)) {
+      if (showScrollbar && _trackVisibility.resolve(states)) {
```
Comment (xu-baolin) [inline packages/flutter/lib/src/material/scrollbar.dart:353]: This is reasonable, maybe we can documentation this behavior.

File: packages/flutter/lib/src/widgets/scrollbar.dart [hunk 2]
```diff
@@ -1199,6 +1206,36 @@ class RawScrollbar extends StatefulWidget {
   /// When null, it will default to the value of [minThumbLength].
   final double? minOverscrollLength;
 
+  /// {@template flutter.widgets.Scrollbar.trackVisibility}
+  /// Indicates that the scrollbar track should be visible.
+  ///
+  /// When true, the scrollbar track will always be visible so long as the thumb
```
Comment (xu-baolin) [inline packages/flutter/lib/src/widgets/scrollbar.dart:1212]: oh, I see this now :)

File: packages/flutter/lib/src/widgets/scrollbar.dart [hunk 3]
```diff
@@ -894,6 +897,10 @@ class RawScrollbar extends StatefulWidget {
          'Scrollbar thumb appearance should only be controlled with thumbVisibility, '
          'isAlwaysShown is deprecated.'
        ),
+       assert(
+         !((thumbVisibility == false || isAlwaysShown == false) && trackVisibility == true),
```
Comment (xu-baolin) [inline packages/flutter/lib/src/widgets/scrollbar.dart:901]: The assertion is a bit unfriendly. In this case, don't draw the track and documentation of this behavior. What do you think?
At the same time, the `Scrollbar` widget does not assert this, right?

Comment (Piinks) [reply packages/flutter/lib/src/widgets/scrollbar.dart:901]: The scrollbar widget does not assert this, but the _MaterialScrollbar and CupertinoScrollbar pass these values to the super class RawScrollbar, so they still get the assertion. 
Here, where the user explicitly says 'show me the track, don't show me the thumb' I think it is reasonable to assert. 

When ScrollbarThemeData.trackVisibility (MaterialStateProperty) is used, and the value changes based on different state combinations, it does not assert and just assumes that if the thumb is not visible then the track should not be.

Comment (xu-baolin) [reply packages/flutter/lib/src/widgets/scrollbar.dart:901]: That makes sense, thanks for explaining.

**Review by xu-baolin — APPROVED** [review]
LGTM

**Pull request merged**